src/v/storage/segment_appender_chunk.h

```diff
@@ -24,11 +25,10 @@
 namespace storage {
 class segment_appender_chunk {
 public:
-    static constexpr const size_t chunk_size = 16_KiB;
-
     explicit segment_appender_chunk(size_t alignment)
       : _alignment(alignment)
-      , _buf(ss::allocate_aligned_buffer<char>(chunk_size, alignment)) {
+      , _chunk_size(config::shard_local_cfg().append_chunk_size())
```

Comment:
Purely subjectively I feel that for `segment_appender_chunk`s (them being data-like disposable objects without much behavior) chunk size should be passed in constructor by whoever uses them and for other cases it's okay to use the value from the config singleton.

Reply:
I'm inclined to agree -- we construct these in batch in chunk_cache so it's easy to fetch the config once.  Updated.